content/providers/01-ai-sdk-providers/22-xai-grok.mdx

```diff
@@ -0,0 +1,104 @@
+---
+title: xAI Grok
+description: Learn how to use xAI Grok.
+---
+
+# Grok Provider
+
+The [xAI Grok](https://x.ai) provider contains language model support for the [xAI API](https://x.ai/api).
+
+## Setup
+
+The xAI Grok provider is available via the `@ai-sdk/grok` module. You can
+install it with
+
+<Tabs items={['pnpm', 'npm', 'yarn']}>
+  <Tab>
+    <Snippet text="pnpm add @ai-sdk/grok" dark />
+  </Tab>
+  <Tab>
+    <Snippet text="npm install @ai-sdk/grok" dark />
+  </Tab>
+  <Tab>
+    <Snippet text="yarn add @ai-sdk/grok" dark />
+  </Tab>
+</Tabs>
+
+## Provider Instance
+
+You can import the default provider instance `xai` from `@ai-sdk/xai`:
+
+```ts
+import { xai } from '@ai-sdk/xai';
+```
+
+If you need a customized setup, you can import `createXai` from `@ai-sdk/xai`
+and create a provider instance with your settings:
+
+```ts
+import { createXai } from '@ai-sdk/xai';
+
+const xai = createXai({
+  // custom settings
+});
+```
+
+You can use the following optional settings to customize the xAI provider instance:
+
+- **baseURL** _string_
+
+  Use a different URL prefix for API calls, e.g. to use proxy servers.
+  The default prefix is `https://api.x.ai/v1`.
+
+- **apiKey** _string_
+
+  API key that is being send using the `Authorization` header. It defaults to
+  the `XAI_API_KEY` environment variable.
+
+- **headers** _Record&lt;string,string&gt;_
+
+  Custom headers to include in the requests.
+
+- **fetch** _(input: RequestInfo, init?: RequestInit) => Promise&lt;Response&gt;_
+
+  Custom [fetch](https://developer.mozilla.org/en-US/docs/Web/API/fetch) implementation.
+  Defaults to the global `fetch` function.
+  You can use it as a middleware to intercept requests,
+  or to provide a custom fetch implementation for e.g. testing.
+
+## Language Models
+
+You can create [xAI models](https://console.x.ai) using a provider instance. The
+first argument is the model id, e.g. `grok-beta`.
+
+```ts
+const model = xai('grok-beta');
+```
+
+### Example
+
+You can use xAI language models to generate text with the `generateText` function:
+
+```ts
+import { xai } from '@ai-sdk/xai';
+import { generateText } from 'ai';
+
+const { text } = await generateText({
+  model: xai('grok-beta'),
+  prompt: 'Write a vegetarian lasagna recipe for 4 people.',
+});
+```
+
+## Model Capabilities
+
+xAI Grok offers [a variety of models with different
+capabilities](https://console.x.ai/) under "View models", including:
+
+| Model       | Image Input         | Object Generation   | Tool Usage          | Tool Streaming      |
+| ----------- | ------------------- | ------------------- | ------------------- | ------------------- |
+| `grok-beta` | <Cross size={18} /> | <Check size={18} /> | <Check size={18} /> | <Check size={18} /> |
```

Comment:
are there any limitations in their tool usage?

Reply:
Not that I know of currently. Tool streaming just means tool calls work with `streamText()`?

Reply:
this is great, it means we can set object generation to `tool` mode with forced function calling